### ISSUE TITLE
[BUG] npm install - Bundled dependencies are not saved to package.json

### ISSUE THREAD

**Issue by CodyEakins** [npm/cli#3174]
### Current Behavior:

If you install any package with the bundled dependency flag:
```
npm install <package-name> --save-bundle
```
... `<package-name>` is not added to the `bundledDependencies` array in your package.json file.

### Expected Behavior:

When installing any package with the bundled dependency flag (`--save-bundle` or `-B`), the package should be installed and added to the `bundledDependencies` array in your package.json file.

### Steps To Reproduce:

1. In this environment, create a new directory, step into it, and run `npm init`.
2. After creating a plain/empty package.json, run `npm install <package-name> --save-bundle`.
3. Open package.json, and discover that the `bundledDependencies` field does not exist, and `<package-name>` is not listed within it.

### Environment:

- OS: Ubuntu 20.10
- Node: 14.15.0
- npm: 7.11.2

### Additional Comments:

Interestingly enough, if you run `npm install --help` on v7.11.2, you'll see:
```
npm install [<@scope>/]<pkg>
npm install [<@scope>/]<pkg>@<tag>
npm install [<@scope>/]<pkg>@<version>
npm install [<@scope>/]<pkg>@<version range>
npm install <alias>@npm:<name>
npm install <folder>
npm install <tarball file>
npm install <tarball url>
npm install <git:// url>
npm install <github username>/<github project>

Options:
[-S|--save|--no-save|--save-prod|--save-dev|--save-optional|--save-peer]
[-E|--save-exact]
```
Notice that the `--save-bundle` flag IS NOT documented. However, if you read the [v7 npm install docs](https://docs.npmjs.com/cli/v7/commands/npm-install), you'll see that the bundled dependency flags and behaviors ARE documented.

**Comment by wraithgar** [comment]
`--save-bundle` is a valid config flag and should be showing up in the help output, so there are two bugs here: the missing behavior and the incorrect help output.

**Comment by CodyEakins** [comment]
> `--save-bundle` is a valid config flag and should be showing up in the help output, so there are two bugs here: the missing behavior and the incorrect help output.

I can write another issue if you'd like. Please make this issue more about the missing behavior; not the documentation issue.

**Comment by wraithgar** [comment]
It is, that is just a note for whoever picks this up.

**Comment by menduz** [comment]
Hello, is there something we can do to help with this issue? It is affecting our operations at the Decentraland Community

**Comment by wraithgar** [comment]
fixed by #4387 (and docs update is in #4388)
